aten/src/ATen/native/native_functions.yaml

```diff
@@ -3051,6 +3051,15 @@
   variants: method
   device_guard: False
 
+- func: vandermonde_matrix(Tensor x) -> Tensor
+  use_c10_dispatcher: unboxed_only
```

Comment:
Why unboxed only?

Reply:
was just mimicking the other window functions. I didn't put much thought into it. I am guessing its not needed?

Reply:
Try full and try supporting Named Tensor. You don't have to add a test in the test suite for named tensor, but you can it locally yourself.